otwcode/otwarchive
# AO3-6085 Fix eager loading for zeitwerk mode

## Issue

https://otwarchive.atlassian.net/browse/AO3-6085

## Purpose

Enable `config.eager_load` in test, but [only for CI environments](https://guides.rubyonrails.org/classic_to_zeitwerk_howto.html#continuous-integration).

Stop using `config.autoload_paths`, which is [not recommended in Rails 6+](https://guides.rubyonrails.org/v6.0/configuring.html#rails-general-configuration).

Fix constant paths to match [Zeitwerk's expectations](https://github.com/fxn/zeitwerk#file-structure):

- Add app/models sub-directories to [`config.eager_load_paths`](https://guides.rubyonrails.org/v6.0/autoloading_and_reloading_constants_classic_mode.html#autoload-paths-and-eager-load-paths), because those constants are already referenced everywhere.

- The live validations plugin is basically a monkey-patch for ActionView, and is not very modular. It's simplest to just move it into config/initializers/monkeypatches, where Zeitwerk does not complain about the file path and constant path mismatch.

- The commentable plugin is written like a concern, so renaming the constants is easy.

Also remove an outdated README (we no longer set up ArchiveConfig in an initializer) and an unused gem initializer.

## Testing Instructions

Test suite should pass. For a quick run locally, use [`bundle exec rails zeitwerk:check`](https://guides.rubyonrails.org/classic_to_zeitwerk_howto.html#check-zeitwerk-compliance-in-the-test-suite).

Site should load.
